scipy/interpolate/_rgi.py

```diff
@@ -286,12 +279,10 @@ def __init__(self, points, values, method="linear", bounds_error=True,
         if self._descending_dimensions:
             self.values = np.flip(values, axis=self._descending_dimensions)
         if self.method == "pchip" and np.iscomplexobj(self.values):
-            msg = ("`PchipInterpolator` only works with real values. Passing "
-                   "complex-dtyped `values` with `method='pchip'` is deprecated "
-                   "and will raise an error in SciPy 1.15.0. If you are trying to "
-                   "use the real components of the passed array, use `np.real` on "
+            msg = ("`PchipInterpolator` only works with real values. If you are trying "
+                   "to use the real components of the passed array, use `np.real` on "
                    "the array before passing to `RegularGridInterpolator`.")
-            warnings.warn(msg, DeprecationWarning, stacklevel=2)
+            raise TypeError(msg)
```

Comment:
do you want to change this to a `ValueError` too?

Reply:
Oops yes!